glean-core/src/metrics/labeled.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::marker::PhantomData;
+
+use crate::metrics::Metric;
+use crate::metrics::MetricType;
+use crate::storage::StorageManager;
+use crate::CommonMetricData;
+use crate::Glean;
+
+#[derive(Debug)]
+pub struct LabeledMetric<T> {
+    meta: CommonMetricData,
+    labels: Option<Vec<String>>,
+    // Type of the underlying metric
+    // We don't hold on to any of these, but we need to be able to create them.
+    typ: PhantomData<T>,
+}
+
+impl<T> LabeledMetric<T>
+where
+    T: MetricType,
+{
+    pub fn new(meta: CommonMetricData, labels: Option<Vec<String>>) -> LabeledMetric<T> {
+        LabeledMetric {
+            meta,
+            labels,
```

Comment:
With the current design, how are we going to cope with dynamic labels?

Reply:
not fleshed out at all.